### PR TITLE
test: analyze JUnit results in PR jobs

### PR DESCRIPTION
**Pull request by epuertat**
Ceph changes brought by https://github.com/ceph/ceph/pull/36771

Fixes: https://tracker.ceph.com/issues/47118
Signed-off-by: Ernesto Puerta <epuertat@redhat.com>

### CODE REVIEW

**Review by LenzGr — APPROVED** [review]
:+1: thanks!

**Review by callithea — APPROVED** [review]

File: ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@
           auto-rebuild: true
       - inject:
           properties-content: |
-            TERM=xterm
+              TERM=xterm-256color
```
Comment (djgalloway) [inline ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml:24]: Why the extra spacing?

Comment (epuertat) [reply ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml:24]: Unintentional... Will fix it. Thanks!

**Comment by djgalloway** [comment]
jenksin test

**Comment by djgalloway** [comment]
jenkins test

File: ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml [hunk 2]
```diff
@@ -78,4 +78,9 @@
           - text:
               credential-id: cd-cypress-record-key
               variable: CYPRESS_RECORD_KEY
-      - ansicolor
+      - ansicolor:
+          colormap: "gnome-terminal"
+
+    publishers:
+      - junit:
+          results: **/reports/*.xml
```
Comment (djgalloway) [inline ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml:86]: I think asterisks need to be in quotes

**Review by djgalloway — CHANGES_REQUESTED** [review]

File: ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml (hunk 2 above)
Comment (epuertat) [reply ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml:86]: You're right! Thanks!

**Review by djgalloway — APPROVED** [review]

**Comment by epuertat** [comment]
@djgalloway as https://github.com/ceph/ceph/pull/36771 has already been merged, if you're ok with this, we can merge it and see how JUnit tests start being reported for Dashboard PRs.

**Comment by epuertat** [comment]
Thanks @djgalloway !

**Comment by epuertat** [comment]
Let's revert this, as it is causing errors in all jobs:

> Recording test results
> Agent went offline during the build
> ERROR: Connection was broken: java.nio.channels.ClosedChannelException
> 	at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer.onReadClosed(ChannelApplicationLayer.java:209)
> 	at org.jenkinsci.remoting.protocol.ApplicationLayer.onRecvClosed(ApplicationLayer.java:221)
> 	at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecvClosed(ProtocolStack.java:816)
> 	at org.jenkinsci.remoting.protocol.FilterLayer.onRecvClosed(FilterLayer.java:288)
> 	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.onRecvClosed(SSLEngineFilterLayer.java:181)
> 	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.switchToNoSecure(SSLEngineFilterLayer.java:283)
> 	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processWrite(SSLEngineFilterLayer.java:503)
> 	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processQueuedWrites(SSLEngineFilterLayer.java:248)
> 	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doSend(SSLEngineFilterLayer.java:200)
> 	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doCloseSend(SSLEngineFilterLayer.java:213)
> 	at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.doCloseSend(ProtocolStack.java:784)
> 	at org.jenkinsci.remoting.protocol.ApplicationLayer.doCloseWrite(ApplicationLayer.java:172)
> 	at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer$ByteBufferCommandTransport.closeWrite(ChannelApplicationLayer.java:314)
> 	at hudson.remoting.Channel.close(Channel.java:1502)
> 	at hudson.remoting.Channel.close(Channel.java:1455)
> 	at hudson.slaves.SlaveComputer.closeChannel(SlaveComputer.java:874)
> 	at hudson.slaves.SlaveComputer.access$100(SlaveComputer.java:110)
> 	at hudson.slaves.SlaveComputer$2.run(SlaveComputer.java:765)
> 	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
> 	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:59)
> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
> 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> 
> Build step 'Publish JUnit test result report' marked build as failure
> ERROR: Step ?Publish Cobertura Coverage Report? failed: no workspace for ceph-pull-requests #60996

**Pull request merged**